### ISSUE TITLE
PCL Profile 259, Version=v4.5

### ISSUE THREAD

**Issue by ThomasLoidolt** [Redth/ZXing.Net.Mobile#258]
Hello,
I 've tried to install the current version of the ZXing.Net.Mobile Package to my PCL but I got the error message: "Could not install package 'ZXing.Net.Mobile 2.0.4.46'. You are trying to install this package into a project that targets '.NETPortable,Version=v4.5,Profile=Profile259', but the package does not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author."

My PCL Project targets the PCL Version 259 and Version 4.5 and as I saw the ZXing.Net.Mobile.Core projects targets it too.

So what might can I do?

**Comment by Giullianogp** [comment]
I have the same problem..

**Comment by Redth** [comment]
So the problem is, ZXing.Net.Mobile doesn't actually support 259 as there's not implementation for every platform in that profile (Windows Phone 8.1 native is not supported, however Windows Phone 8.1 Silverlight is).

I guess I could just ship the Core dll for the platforms that aren't supported, however I'm not exactly sure what that would help achieve (you'd end up getting a reference to the _reference_ PCL for ZXing.Net.Mobile which is just a bunch of stubs that throw exceptions).

I welcome more discussion on this, but for now I see no reason why artificially supporting 259 is a good idea...

**Comment by ThomasLoidolt** [comment]
So what I did, because at the moment I only use it for WP81 Silverlight, is to pull it from the repo compile the core, portable and WP8 project and added it manually to the Core and WP81 project of my concrete app. 
That worked for me!

**Comment by faceoffers28** [comment]
Same problem here. I would love to use this component. I added the DLL to the IOs Project as a Reference. The app compiles and displays the overlay page, but crashes when I add this line to AppDelegate.cs. global::ZXing.Net.Mobile.Forms.iOS.Platform.Init ();  I have not tested Android or Windows Phone 8.1 Silverlight.

**Comment by faceoffers28** [comment]
I finally got this working in my Xamarin.Forms PCL 259 app by doing the following: I added package 2.0.4.46 to my Portable app, using Nuget. I then manually referenced the only DLL (ZXing.Net.Mobile.dll) in the ZXIng.Net.Mobile.Forms/Lib/Portable- directory. I then manually referenced the other 2 DLLs (ZXing.Net.Mobile.Core.dll and zxing.portable.dll)) in the ZXIng.Net.Mobile/Lib/Portable- directory. I then added package 2.0.4.46 as a Component in my IOs app and it worked. I can't get it to work in my Android app right now due to the Xamarin.Forms 2.2 issues and the Android support libraries, but hopefully Zing.Net.Mobile.Forms will be updated soon.
